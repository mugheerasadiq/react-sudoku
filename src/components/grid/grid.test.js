import React from "react";
import renderer from "react-test-renderer";

import Grid from "./Grid.index";

const gridState = [
  [0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0],
];

it("All props are passed", () => {
  const tree = renderer
    .create(
      <Grid
        setLoader={() => {}}
        setGrid={() => {}}
        grid={gridState}
        level={"easy"}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
