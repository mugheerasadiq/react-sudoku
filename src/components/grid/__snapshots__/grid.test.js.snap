// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All props are passed 1`] = `
<table>
  <tbody>
    <tr>
      <td>
        <input
          name="0,0"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#90CAF9",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="0,1"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#90CAF9",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="0,2"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#90CAF9",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="0,3"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#1DE9B6",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="0,4"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#1DE9B6",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="0,5"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#1DE9B6",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="0,6"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#FFAB91",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="0,7"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#FFAB91",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="0,8"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#FFAB91",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
    </tr>
    <tr>
      <td>
        <input
          name="1,0"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#90CAF9",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="1,1"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#90CAF9",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="1,2"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#90CAF9",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="1,3"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#1DE9B6",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="1,4"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#1DE9B6",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="1,5"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#1DE9B6",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="1,6"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#FFAB91",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="1,7"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#FFAB91",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="1,8"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#FFAB91",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
    </tr>
    <tr>
      <td>
        <input
          name="2,0"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#90CAF9",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="2,1"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#90CAF9",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="2,2"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#90CAF9",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="2,3"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#1DE9B6",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="2,4"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#1DE9B6",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="2,5"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#1DE9B6",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="2,6"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#FFAB91",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="2,7"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#FFAB91",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="2,8"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#FFAB91",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
    </tr>
    <tr>
      <td>
        <input
          name="3,0"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#D1C4E9",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="3,1"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#D1C4E9",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="3,2"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#D1C4E9",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="3,3"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#FFF59D",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="3,4"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#FFF59D",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="3,5"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#FFF59D",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="3,6"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#A5D6A7",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="3,7"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#A5D6A7",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="3,8"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#A5D6A7",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
    </tr>
    <tr>
      <td>
        <input
          name="4,0"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#D1C4E9",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="4,1"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#D1C4E9",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="4,2"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#D1C4E9",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="4,3"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#FFF59D",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="4,4"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#FFF59D",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="4,5"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#FFF59D",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="4,6"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#A5D6A7",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="4,7"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#A5D6A7",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="4,8"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#A5D6A7",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
    </tr>
    <tr>
      <td>
        <input
          name="5,0"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#D1C4E9",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="5,1"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#D1C4E9",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="5,2"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#D1C4E9",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="5,3"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#FFF59D",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="5,4"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#FFF59D",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="5,5"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#FFF59D",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="5,6"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#A5D6A7",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="5,7"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#A5D6A7",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="5,8"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#A5D6A7",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
    </tr>
    <tr>
      <td>
        <input
          name="6,0"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#80CBC4",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="6,1"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#80CBC4",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="6,2"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#80CBC4",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="6,3"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#F48FB1",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="6,4"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#F48FB1",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="6,5"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#F48FB1",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="6,6"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#81D4FA",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="6,7"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#81D4FA",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="6,8"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#81D4FA",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
    </tr>
    <tr>
      <td>
        <input
          name="7,0"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#80CBC4",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="7,1"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#80CBC4",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="7,2"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#80CBC4",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="7,3"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#F48FB1",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="7,4"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#F48FB1",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="7,5"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#F48FB1",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="7,6"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#81D4FA",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="7,7"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#81D4FA",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="7,8"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#81D4FA",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
    </tr>
    <tr>
      <td>
        <input
          name="8,0"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#80CBC4",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="8,1"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#80CBC4",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="8,2"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#80CBC4",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="8,3"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#F48FB1",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="8,4"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#F48FB1",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="8,5"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#F48FB1",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="8,6"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#81D4FA",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="8,7"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#81D4FA",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
      <td>
        <input
          name="8,8"
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#81D4FA",
              "fontSize": "3em",
              "fontWeight": "150",
            }
          }
          value=""
        />
      </td>
    </tr>
  </tbody>
</table>
`;
